AnnService/src/Helper/AsyncFileReader.cpp

```diff
@@ -82,39 +82,48 @@ namespace SPTAG {
             }
         }
 #else
-        void BatchReadFileAsync(std::vector<std::shared_ptr<Helper::DiskPriorityIO>>& handlers, AsyncReadRequest* readRequests, int num)
+        using BatchOp = bool (DiskPriorityIO::*)(AsyncReadRequest*, uint32_t);
+        void CallOnAppropriateBatch(std::vector<std::shared_ptr<Helper::DiskPriorityIO>>& handlers, AsyncReadRequest* readRequests, int num, BatchOp f)
         {
-            
             if (handlers.size() == 1) {
-                handlers[0]->BatchReadFile(readRequests, num);
+                (handlers[0].get()->*f)(readRequests, num);
             }
-            else {            
+            else {
                 int currFileId = 0, currReqStart = 0;
                 for (int i = 0; i < num; i++) {
                     AsyncReadRequest* readRequest = &(readRequests[i]);
                     if (readRequest->m_readSize == 0) continue;
 
                     int fileid = (readRequest->m_status >> 16);
                     if (fileid != currFileId) {
-                        handlers[currFileId]->BatchReadFile(readRequests + currReqStart, i - currReqStart);
+                        (handlers[currFileId].get()->*f)(readRequests + currReqStart, i - currReqStart);
                         currFileId = fileid;
                         currReqStart = i;
                     }
                 }
                 if (currReqStart < num) {
-                    handlers[currFileId]->BatchReadFile(readRequests + currReqStart, num - currReqStart);
+                    (handlers[currFileId].get()->*f)(readRequests + currReqStart, num - currReqStart);
                 }
             }
+        }
+
+        void BatchReadFileAsync(std::vector<std::shared_ptr<Helper::DiskPriorityIO>>& handlers, AsyncReadRequest* readRequests, int num)
+        {
+            
+            CallOnAppropriateBatch(handlers, readRequests, num, &DiskPriorityIO::BatchReadFile);
 
             for (int i = 0; i < num; i++) {
                 AsyncReadRequest* readRequest = &(readRequests[i]);
                 if (readRequest->m_readSize == 0) continue;
-                readRequest->m_readSize = 0;
```

Comment:
We need to clean readSize here since we use readSize to judge whether we need to issue a read request or not.